### PR TITLE
build!: compile and run CI on stable, fix #1587

### LINKED ISSUE #1587: Run CI on stable Rust rather than latest nightly
Some of our crates (`hydroflow` at least) are expected to compile on stable Rust. We should run these compilation checks in CI.

In addition, running lints on `latest-nightly` seems to not be worth the effort, since it just fails every build on `main` and we pin our nightly version in the repo and templates anyways. So we can reduce the CI burden there.

### PR DESCRIPTION
**Pull request by MingweiSamuel**
fix #1587

* use `nightly` cfg set in `build.rs`es instead of `feature = "nightly"` which could be incorrectly set
* run CI on stable rust instead of latest nightly

BREAKING CHANGE: `hydroflow_lang::Diagnostic::emit` replaced by `hydroflow_lang::Diagnostic::try_emit` to work on both stable and nightly, returning a result.

### CODE REVIEW

File: hydroflow_plus/tests/compile_fail.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#![cfg(nightly)]
```
Comment (shadaj) [inline hydroflow_plus/tests/compile_fail.rs:1]: Oh does trybuild not work on nightly??

Comment (MingweiSamuel) [reply hydroflow_plus/tests/compile_fail.rs:1]: So interesting setup, for nightly dev we have `.cargo/config.toml` set up to set nightly flags namely `-Zproc-macro-backtrace`, but `-Z` unstable options do not work on nightly so cargo immediately exits. We can override this on stable by setting `RUSTFLAGS=""`. When trybuild runs it tries to run with the `.cargo/config.toml` and compilation fails. There is a way to work around it, but in general the error messages will be different between nightly and stable, so we should just stick with one to snapshot test, and I went with nightly

Comment (shadaj) [reply hydroflow_plus/tests/compile_fail.rs:1]: Could we just... not put `-Zproc-macro-backtrace`? I feel like that was important when we had more bugs in our macros but now they're mostly ironed out.

Comment (MingweiSamuel) [reply hydroflow_plus/tests/compile_fail.rs:1]: Yeah that would work too....

**Pull request closed without merging**